smartsim/_core/shell/shellLauncher.py

```diff
@@ -0,0 +1,65 @@
+# BSD 2-Clause License
+#
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+from __future__ import annotations
+
+import typing as t
+
+from smartsim.types import LaunchedJobID
+
+from smartsim.log import get_logger
+
+if t.TYPE_CHECKING:
+    from typing_extensions import Self
+    from smartsim.experiment import Experiment
+
+import subprocess as sp
+
+from smartsim._core.utils import helpers
+from smartsim._core.dispatch import dispatch
+
+logger = get_logger(__name__)
+
+class ShellLauncher:
+    """Mock launcher for launching/tracking simple shell commands"""
+
+    def __init__(self) -> None:
+        self._launched: dict[LaunchedJobID, sp.Popen[bytes]] = {}
+
+    def start(self, command: t.Sequence[str]) -> LaunchedJobID:
+        id_ = dispatch.create_job_id()
+        exe, *rest = command
+        # pylint: disable-next=consider-using-with
+        self._launched[id_] = sp.Popen((helpers.expand_exe_path(exe), *rest))
+        return id_
+
+    @classmethod
+    def create(cls, _: Experiment) -> Self:
+        return cls()
+    
+
+# <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
```

Comment:
Nit: Feel free to drop this comment since it's no longer delimiting the end of a block!

Reply:
removed